File: app/src/main/java/com/example/polyquicktrade/HomeViewModel.java
package com.example.polyquicktrade;

import com.example.polyquicktrade.pojo.Enquiry;
import com.google.android.gms.tasks.OnFailureListener;
import com.google.android.gms.tasks.OnSuccessListener;
import com.google.firebase.firestore.DocumentReference;
import com.google.firebase.firestore.FirebaseFirestore;

import java.util.Calendar;
import java.util.Date;

import androidx.annotation.NonNull;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class HomeViewModel extends ViewModel {

    public MutableLiveData<Enquiry> getEnquiry() {
        return enquiryMutableLiveData;
    }

    private MutableLiveData<Enquiry>  enquiryMutableLiveData;

    public HomeViewModel() {
        this.enquiryMutableLiveData = new MutableLiveData<>();
    }

    public void uploadEnquireToFirebase(FirebaseFirestore db, final Enquiry enquiry){

        Date date = new Date(enquiry.getCreationTime());
        Calendar cal = Calendar.getInstance();
        cal.setTime(date);


        int month = cal.get(Calendar.MONTH);
        int week = cal.get(Calendar.WEEK_OF_MONTH);
        int day = cal.get(Calendar.DAY_OF_WEEK);


        //Database Storage: Enquiries C-> Month D-> Week C -> Enquiry D
        DocumentReference enqRef = db
                .collection("Enquiry")
                .document("Month_" + month)
                .collection("enquiry_documents")
                .document();

        enquiry.setId(enqRef.getId());
        enqRef.set(enquiry).addOnSuccessListener(new OnSuccessListener<Void>() {
            @Override
            public void onSuccess(Void aVoid) {
                enquiryMutableLiveData.setValue(enquiry);

            }
        }).addOnFailureListener(new OnFailureListener() {
            @Override
            public void onFailure(@NonNull Exception e) {

            }
        });




    }
}

File: app/src/main/java/com/example/polyquicktrade/HomeViewModel.kt
package com.example.polyquicktrade

import com.example.polyquicktrade.pojo.Enquiry.creationTime
import com.example.polyquicktrade.pojo.Enquiry.id
import androidx.lifecycle.ViewModel
import androidx.lifecycle.MutableLiveData
import com.example.polyquicktrade.pojo.Enquiry
import com.google.firebase.firestore.FirebaseFirestore
import com.google.android.gms.tasks.OnSuccessListener
import com.google.android.gms.tasks.OnFailureListener
import java.util.*

class HomeViewModel : ViewModel() {
    val enquiry: MutableLiveData<Enquiry>
    fun uploadEnquireToFirebase(db: FirebaseFirestore, enquiry: Enquiry) {
        val date = Date(enquiry.creationTime)
        val cal = Calendar.getInstance()
        cal.time = date
        val month = cal[Calendar.MONTH]
        val week = cal[Calendar.WEEK_OF_MONTH]
        val day = cal[Calendar.DAY_OF_WEEK]


        //Database Storage: Enquiries C-> Month D-> Week C -> Enquiry D
        val enqRef = db
            .collection("Enquiry")
            .document("Month_$month")
            .collection("enquiry_documents")
            .document()
        enquiry.id = enqRef.id
        enqRef.set(enquiry).addOnSuccessListener { enquiry.setValue(enquiry) }
            .addOnFailureListener { }
    }

    init {
        enquiry = MutableLiveData()
    }
}